content/chainguard/chainguard-images/getting-started/nginx/index.md

```diff
@@ -0,0 +1,262 @@
+
+```shell
+curl -O https://raw.githubusercontent.com/chainguard-dev/edu-images-demos/734e6171ee69f0e0dbac95e6ebc1759ac18bf00a/nginx/data/inky.png
+```
+
+Now, return to the `nginxdemo` directory you made earlier. Here we will create the `nginx.conf` configuration file used by nginx to run the local web server. We will demonstrate this using `nano`.
+
+```shell
+nano nginx.conf
+```
+
+Copy the following code into the configuration file you created. In the `location` directive, be sure to update the configuration to reference the path from root to the `data` directory you created in a previous step.
+
+```nginx
+worker_processes  1;
+
+events {
+    worker_connections  1024;
+}
+
+http {
+    include       mime.types;
+    default_type  application/octet-stream;
+
+    sendfile        on;
+
+    keepalive_timeout  65;
+
+    server {
+
+        listen 1313;
+        server_name localhost;
+        charset koi8-r;
+
+        location / {
+            root /Users/username/nginxdemo/data; # Update the file path for your system
+
+        }
+
+    }
+
+    include servers/*;
+
+}
+```
+
+Once you are finished editing the configuration file, save and close it.
+
+Create a new file named `mime.types` using a text editor of your choice. We will use `nano` to demonstrate.
+
+```shell
+nano mime.types
+```
+
+Copy and paste the following code into your `mime.types` file. This file will allow nginx to handle the HTML, CSS, and png files we created when rendering the webserver. 
+
+```nginx
+types {
+    text/html                                        html htm shtml;
+    text/css                                         css;
+    image/png                                        png;
+}
+```
+
+Save and close the `mime.types` file when you are finished editing it.
+
+Next, copy the filepath to the `nginx.conf` file you created earlier. Replacing the example path with this copied path, execute the following command to initialize the nginx server:
+
+```shell
+nginx -c /Users/username/nginxdemo/nginx.conf
+```
+
+To view the HTML content, navigate to `localhost:8080` in your web browser of choice. You should see a simple landing page with a picture of Inky and a fun fact about the Wolfi octopus.
+
+If you make any changes to the files nginx is serving, run `nginx -s reload` in your terminal to allow the changes to render. When you are finished with your application, run `nginx -s quit` to allow nginx to safely shut down.
+
+## Step 2: Creating the Dockerfile
+
+We will now use a Dockerfile to build an image containing the demo application. 
+
+In the `nginxdemo` directory, create the `Dockerfile` with the text editor of your choice. We will use `nano`:
+
+```shell
+nano Dockerfile
+```
+The following Dockerfile will:
+
+1. Start a new build based on the `cgr.dev/chainguard/nginx:latest` image;
+2. Expose port 8080 in the container for nginx to listen on;
```

Comment:
This is pretty fiddly and annoying, but the Dockerfile isn't actually exposing the port here, the expose keyword just creates metadata. It's the -p flag that actually does the work. (Try removing the expose line in the Dockerfile and it should still have the same behavior as a test.) Basically just saying not to say that the Dockerfile will expose the port, you could change the line to say something about metadata, take it out, etc, up to you.